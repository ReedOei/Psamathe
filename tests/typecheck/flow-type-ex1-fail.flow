contract Main {}
;

var x : ! nat := 6
var y : ! string := "test"

x --> y

