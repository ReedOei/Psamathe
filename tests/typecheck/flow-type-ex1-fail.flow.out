  <generatedTop>
    <common>
      <k>
        y accepts ! nat ~> update ( y , typeof ( y ) +Ty ! ) ~> drop ( y ) ~> drop ( x ) ~> .
      </k>
      <contracts>
        SetItem ( Main )
      </contracts>
      <fields>
        Main |-> .Set
      </fields>
      <constructors>
        Main |-> constructor ( .VarDefs , pass )
      </constructors>
      <functions>
        Main |-> .Set
      </functions>
      <types>
        Main |-> SetItem ( type ( Main , SetItem ( asset )
        SetItem ( consumable ) , Main ) )
        address |-> SetItem ( type ( address , SetItem ( consumable )
        SetItem ( nonfungible ) , address ) )
        bool |-> SetItem ( type ( bool , SetItem ( consumable )
        SetItem ( nonfungible ) , bool ) )
        nat |-> SetItem ( type ( nat , SetItem ( consumable )
        SetItem ( fungible ) , nat ) )
        string |-> SetItem ( type ( string , SetItem ( consumable )
        SetItem ( nonfungible ) , string ) )
      </types>
      <freshCounter>
        0
      </freshCounter>
    </common>
    <typeEnv>
      x |-> empty nat
      y |-> ! string
    </typeEnv>
  </generatedTop>
#And
  {
    ?Q
  #Equals
    !
  }
#And
  {
    ?R
  #Equals
    every
  }
#And
  {
    ?Tau
  #Equals
    ! nat
  }
